libs/store/src/lib/infrastructure/persistence/store/entity-collection-service-base.ts

```diff
@@ -0,0 +1,252 @@
+import { BehaviorSubject, merge, Observable, Subject, ReplaySubject, zip } from 'rxjs';
+import { EntityCollection, EntityCollectionService as NgrxEntityCollectionService, EntityServices } from '@ngrx/data';
+import { Store } from '@ngrx/store';
+import { map, switchMap, takeUntil } from 'rxjs/operators';
+
+import { QuerySnapshotModel } from '../query/query';
+import { BaseEntity } from './entity-server/interfaces';
+import { EntityMetaOptions, EntityType, IdentityKey } from '../../../domain/public_api';
+import { EntityMetaOptionsService } from '../utils/entity-options.service';
+import { EntityCollectionService } from './entity-collection-service';
+import { EntityCollectionsService } from './entity-collections-service';
+import { ChainingPolicy, ChainingPolicyFieldOptions } from '../../../domain/chaining-policy';
+
+/**
+ * Default EntityCollectionService implementation.
+ *
+ * The main idea of this service
+ *
+ * That is a wrapper for NgRx EntityCollectionService
+ * it delegates all heavy work to ngrx-data.
+ *
+ * Also this service is responsible to perform request chaining
+ *
+ * Also this service is responsible to chain requests for sub resources
+ *
+ */
+export class DefaultEntityCollectionService<T extends BaseEntity> implements EntityCollectionService<T> {
+    readonly name: string;
+
+    get collection$(): Observable<EntityCollection<T>> | Store<EntityCollection<T>> {
+        return this.entityCollectionService.collection$;
+    }
+
+    get count$(): Observable<number> | Store<number> {
+        return this.entityCollectionService.count$;
+    }
+
+    get entities$(): Observable<T[]> | Store<T[]> {
+        return this.entityCollectionService.entities$;
+    }
+
```

Comment:
Do we want to expose other selectors here like `errors$`, `loading$` and `loaded$`?

Reply:
We said that we will expose this into EntityStore no?

Reply:
added `errors$, loading, loaded$` properties, so it can be used by store